backend/dataall/api/Objects/ShareObject/resolvers.py

```diff
@@ -375,3 +375,37 @@ def list_shares_in_my_outbox(context: Context, source, filter: dict = None):
             data=filter,
             check_perm=None,
         )
+
+
+def update_share_request_purpose(context: Context, source, shareUri: str = None, requestPurpose: str = None):
+    with context.engine.scoped_session() as session:
+        share = db.api.ShareObject.get_share_by_uri(session, shareUri)
+        userRoleForShareObject = resolve_user_role(context, source=share)
+        print(userRoleForShareObject)
+        if userRoleForShareObject != ShareObjectPermission.Requesters.value:
+            raise db.exceptions.ResourceUnauthorized(
+                username=context.username,
+                action="UPDATE_SHARE_REQUEST_REASON",
+                resource_uri=share.shareUri,
+            )
+
+        return db.api.ShareObject.update_share_request_purpose(
+            session=session,
+            username=context.username,
+            groups=context.groups,
+            uri=shareUri,
+            data={"requestPurpose": requestPurpose},
+            check_perm=None,
+        )
+
+
+def update_share_reject_purpose(context: Context, source, shareUri: str = None, rejectPurpose: str = None):
+    with context.engine.scoped_session() as session:
+        return db.api.ShareObject.update_share_reject_purpose(
```

Comment:
We could do it the same way as for updating share request purpose - I wanted to use the decorator `@has_resource_perm(permissions.REJECT_SHARE_OBJECT)` when possible which we can for updating reject purpose because Approver groups are the only group who are given the ability to reject/approve the share